aten/src/ATen/TensorUtils.cpp

```diff
@@ -68,6 +68,14 @@ void checkAllContiguous(CheckedFrom c, at::ArrayRef<TensorArg> ts) {
 }
 
 void checkSize(CheckedFrom c, const TensorGeometryArg& t, IntArrayRef sizes) {
+  checkDim(c, t, sizes.size());
+  TORCH_CHECK(
+    c10::asIntArrayRefSlow(t->sizes()).equals(sizes),
+    "Expected tensor of size ", sizes, ", but got tensor of size ", t->sizes(),
+    " for ", t, " (while checking arguments for ", c, ")");
+}
+
+void checkSize(CheckedFrom c, const TensorGeometryArg& t, c10::SymIntArrayRef sizes) {
```

Comment:
Another way to solve the ambiguous initializer list problem is to just name this `checkSize_symint`, which isn't a big deal because you're not calling this from templates